src/client/components/widgets/GoDirectoryInput/index.tsx

```diff
@@ -0,0 +1,314 @@
+  onClearQuery?: () => void
+  onKeyPress?: (e: React.KeyboardEvent<HTMLDivElement>) => void
+  query: string
+  getFile:(queryFile: string) => void
+  getState:(queryState: string) => void
+  getEmail:(queryEmail: string) => void
+  setDisablePagination:(disablePagination:boolean) => void
+  onApply: () => void
+  onReset: () => void
+}
+
+const useStyles = makeStyles((theme) =>
+  createStyles({
+    root: {
+      width: '100%',
+      height: '44px',
+      [theme.breakpoints.up('md')]: {
+        height: '70px',
+      },
+    },
+    searchTextField: {
+      width: '100%',
+      height: '100%',
+      '& input::-webkit-search-decoration, & input::-webkit-search-cancel-button, & input::-webkit-search-results-button, & input::-webkit-search-results-decoration': {
+        display: 'none',
+      },
+    },
+    searchInput: {
+      height: '100%',
+      background: 'white',
+      boxShadow: '0px 0px 30px rgba(0, 0, 0, 0.25)',
+      borderRadius: '5px',
+      border: 0,
+      paddingRight: 0,
+      paddingLeft: 0,
+      [theme.breakpoints.up('md')]: {
+        paddingRight: theme.spacing(2),
+      },
+    },
+    searchInputNested: {
+      [theme.breakpoints.up('md')]: {
+        fontSize: '1rem',
+      },
+    },
+    searchInputIcon: {
+      marginTop: '6px',
+      marginLeft: theme.spacing(2),
+      marginRight: theme.spacing(1),
+      [theme.breakpoints.up('md')]: {
+        marginLeft: theme.spacing(4),
+        marginRight: theme.spacing(2.5),
+      },
+    },
+    searchbar:{
+      display: 'contents'
+    },
+    searchOptionsButton: {
+      padding: theme.spacing(0.75),
+      marginRight: theme.spacing(1.5),
+      [theme.breakpoints.up('md')]: {
+        marginRight: theme.spacing(2.5),
+        padding: theme.spacing(1.5),
+      },
+    },
+    closeButton: {
+      padding: theme.spacing(0.75),
+      [theme.breakpoints.up('md')]: {
+        padding: theme.spacing(1.5),
+      },
+    },
+    sortPanel: {
+      width: theme.spacing(50),
+      right: 0,
+      left: 'auto',
+    },
+    filterButton: {
+      height: '100%',
+      paddingLeft: '30px',
+      paddingRight: '30px',
+      borderRadius: '0px'
+    },
+    filterIcon: {
+      paddingLeft: theme.spacing(1.5),
+      verticalAlign: 'middle',
+      [theme.breakpoints.down('sm')]: {
+        paddingLeft: theme.spacing(0.0),
+      },
+    },
+    buttonWrapper: {
+      width: 'auto',
+      display: 'inline-block',
+      [theme.breakpoints.down('sm')]: {
+        display: 'inline-flex',
+      },
+    },
+    labelWrapper: {
+      verticalAlign: 'middle',
+    },
+  }),
+)
+
+const GoDirectoryInput: FunctionComponent<GoSearchInputProps> = ({
+  showAdornments,
+  query,
+  onQueryChange = () => {},
+  onSortOrderChange = () => {},
+  onClearQuery = () => {},
+  onKeyPress = () => {},
+  getFile = () => {},
+  getState = () => {},
+  getEmail = () => {},
+  onApply = () => {},
+  onReset = () => {},
+  setDisablePagination = () => {},
```

Comment:
you might be able to define a no-op function and just use it for all of these defaults(?)

`const noOp = () => {}`